### PR TITLE
Added better SSL certificate support for yoshi local server

### PR DESCRIPTION
**Pull request by amiryonatan**
### 🔦 Summary
Added a new certificate to `yoshi` with `altNames`.
This will allow everyone to work with the same certificate, and once trusted, all local SSL communication will be trusted.

### CODE REVIEW

**Comment by amiryonatan** [comment]
@ronami let me know what u think.
Thanks!

**Comment by yurynix** [comment]
@amiryonatan thanks for working on that!

IMO, we should add fields that would make it clear where this certificate is came from,
currently it's not the case:

```
openssl x509 -in server.crt -text         

Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number: 12090694944137314152 (0xa7cac6fe10ea7f68)
    Signature Algorithm: sha256WithRSAEncryption
        Issuer: CN=localhost
        Validity
            Not Before: May 22 13:21:30 2019 GMT
            Not After : May 20 13:21:30 2024 GMT
        Subject: CN=localhost
        Subject Public Key Info:
            Public Key Algorithm: rsaEncryption
                Public-Key: (2048 bit)
                Exponent: 65537 (0x10001)
        X509v3 extensions:
            X509v3 Subject Alternative Name: 
                DNS:localhost, DNS:local.wix.com

```

So maybe something like
```
[ dn ]
C=IL
ST=Tel Aviv
L=Tel Aviv
O=FED Infra
OU=Yoshi CDN
emailAddress=devex@wix.com
CN = localhost
```

Also we shouldn't have one certificate that is trusted by all users and it's private key exposed on the internet (this repo), `yoshi` should generate a new cert/key pair on each machine.

IMO, the path that we should take is:
1. on startup check if we have cert, if we don't we generate it and save in somewhere like `~/.yoshi-certs`
2. If user has certs generated, we check if user trusts it via:
```
security verify-cert -c ~/.yoshi-certs/server.cert
```
3. If user doesn't trust it, we ask him if he wants to, and add it via:
```
sudo security add-trusted-cert -d -r trustRoot -k /Library/Keychains/System.keychain ~/.yoshi-certs/server.cert

```

File: packages/yoshi/src/tasks/cdn/certificate.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  };
+};
+
+const createAndTrustCertificate = () => {
+  if (!fs.existsSync(certsFullDirName)) {
+    fs.mkdirSync(certsFullDirName);
+  }
+
+  if (!fs.existsSync(certFile)) {
+    const configPath = path.join(
```
Comment (amiryonatan) [inline packages/yoshi/src/tasks/cdn/certificate.js]: I don't know what's the correct way to build this path

File: packages/yoshi/src/tasks/cdn/certificate.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      `openssl req -config ${configPath} -new -x509 -sha256 -newkey rsa:2048 -nodes -keyout ${keyFile} -days 1460 -out ${certFile}`,
+    );
+  }
+
+  const isTrusted = isCertificateTrusted(certFile);
+  if (!isTrusted) {
+    console.log(`An SSL certificate was generated, and must be trusted.`);
+    console.log(`Please enter your sudo password to trust it.`);
+    execSync(
+      `sudo security add-trusted-cert -d -r trustRoot -k /Library/Keychains/System.keychain ${certFile}`,
```
Comment (amiryonatan) [inline packages/yoshi/src/tasks/cdn/certificate.js]: Supporting only mac machines now

**Comment by amiryonatan** [comment]
@yurynix @ronami 
Thanks for the comments.
Please take a look now :)

**Comment by ronami** [comment]
Putting this here for future reference: https://github.com/davewasmer/devcert.

**Pull request closed without merging**